cpp/libs/gst_nmos_sender_plugin/gst_nmos_sender_plugin.cpp

```diff
@@ -185,32 +190,97 @@ static gboolean gst_nmossender_sink_event(GstPad* pad, GstObject* parent, GstEve
         if(caps)
         {
             gchar* caps_str = gst_caps_to_string(caps);
-            g_print("\n\nReceived CAPS from upstream: %s\n", caps_str);
+            // DEBUG
+            // g_print("\n\nReceived CAPS from upstream: %s\n", caps_str);
 
-            for(guint i = 0; i < gst_caps_get_size(caps); i++)
+            const GstStructure* structure = gst_caps_get_structure(caps, 0);
+            const gchar* media_type       = gst_structure_get_name(structure);
+
+            if(g_strcmp0(media_type, "audio/x-raw") == 0)
+            {
+                // DEBUG
+                // g_print("Audio caps detected\n");
+                // GST_INFO_OBJECT(self, "Audio caps detected");
+                //
+                if(!gst_element_link(self->queue, self->audio_payloader))
+                {
+                    GST_ERROR_OBJECT(self, "Failed to link queue to audio_payloader");
+                    return FALSE;
+                }
+                if(!gst_element_link(self->audio_payloader, self->udpsink))
+                {
+                    GST_ERROR_OBJECT(self, "Failed to link audio_payloader to udpsink");
+                    return FALSE;
+                }
+                self->caps            = gst_caps_ref(caps);
+                self->config.is_audio = TRUE;
+                for(guint i = 0; i < gst_caps_get_size(caps); i++)
+                {
+                    gint rate, channels;
+                    const gchar* format;
+                    if(gst_structure_get_int(structure, "rate", &rate))
+                    {
+                        self->config.audio_sender_fields.sampling_rate = rate;
+                    }
+                    if(gst_structure_get_int(structure, "channels", &channels))
+                    {
+                        self->config.audio_sender_fields.number_of_channels = channels;
+                    }
+                    if((format = gst_structure_get_string(structure, "format")))
+                    {
+                        g_free(self->config.audio_sender_fields.format);
```

Comment:
Instead of having to deal with strdup and frees explicitly, why don't you change the config struct so that it owns std::string items instead of gchar*?